pkg/apis/config/types.go

```diff
@@ -51,6 +51,8 @@ type ControllerConfiguration struct {
 
 	// ImagePullSecret provides an opportunity to inject an image pull secret into the resource deployments
 	ImagePullSecret *ImagePullSecret
+
+	HTTPSToApiserverOnly bool
```

Comment:
This flag should be located in the shoot spec, so that it can be configured for each shoot cluster individually.